Paintroid/src/main/res/layout/dialog_pocketpaint_text_tool.xml

```diff
@@ -125,18 +126,38 @@
 
         <View style="@style/PocketPaintDividerHorizontal"/>
 
-        <EditText
-            android:id="@+id/pocketpaint_text_tool_dialog_input_text"
+
+        <LinearLayout
+            android:orientation="horizontal"
             android:layout_width="match_parent"
+            android:weightSum="4.0"
             android:layout_height="wrap_content"
-            android:inputType="textMultiLine"
-            android:layout_marginStart="30dp"
-            android:layout_marginEnd="30dp"
-            android:layout_marginTop="10dp"
-            android:layout_marginBottom="4dp"
-            android:textCursorDrawable="@null"
-            android:hint="@string/text_tool_dialog_input_hint"
-            android:imeOptions="flagNoExtractUi"/>
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="16dp"
+            android:layout_marginTop="4dp"
+            android:layout_marginBottom="10dp"
+            android:gravity="center">
+
+            <EditText
+                android:id="@+id/pocketpaint_text_tool_dialog_input_text"
+                android:layout_width="0dp"
+                android:layout_height="match_parent"
+                android:hint="@string/text_tool_dialog_input_hint"
+                android:imeOptions="flagNoExtractUi"
+                android:inputType="textMultiLine"
+                android:textCursorDrawable="@null"
+                android:layout_weight="3.5"
+                />
+
+            <ImageButton
+                android:id="@+id/ic_pocketpaint_text_tool_confirm_button"
+                style="@style/PocketPaintSelectableButton"
+                android:layout_width="0dp"
+                android:layout_weight="0.4"
```

Comment:
If you remove the `layout_weight` and set the `layout_width` to `wrap_content` you get a natural width for the button.
You can then further just change the `layout_weight` of the `EditText` above to `="1"` - Android will then assign all the remaining horizontal space to the `EditText`